### PR TITLE
Add concat_text component

### PR DESCRIPTION
**Pull request by tverbeiren**

### CODE REVIEW

File: src/concat_text/config.vsh.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: concat_text
+description: |
+  concatenate a number of text files, handle gzipped text files gracefully and
```
Comment (DriesSchaumont) [inline src/concat_text/config.vsh.yaml:3]:
```suggestion
  Concatenate a number of text files, handle gzipped text files gracefully and
```

File: src/concat_text/config.vsh.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    arguments:
+      - name: --input
+        description: A list of (gzipped) text files.
+        type: file
+        multiple: true
+        required: true
+        example: input?.txt.gz
+  - name: Output arguments
+    arguments:
+      - name: "--zip_output"
```
Comment (DriesSchaumont) [inline src/concat_text/config.vsh.yaml:18]: I would propose to create an argument `--compression` in order to specify the compression algorithm to use.  This allows other algorithms to be added in the future

If we choose not to, `zip_output` is also confusing; I would use `gzip_output`

File: src/concat_text/script.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -e
```
Comment (DriesSchaumont) [inline src/concat_text/script.sh:3]: `set- e` has some problems: https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html
```
The shell does not exit if the command that fails is part of the command list immediately following a while or until keyword, part of the test in an if statement, part of any command executed in a && or || list except the command following the final && or ||, any command in a pipeline but the last, or if the command’s return status is being inverted with !
```
So if using `a && b`, and a fails, the script will not exit.

I think its better to use `set -euo pipefail` or similar

File: src/concat_text/script.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
```
Comment (DriesSchaumont) [inline src/concat_text/script.sh:1]: I prefer this:
```suggestion
#!/usr/bin/env bash
```
But not sure if it works on alpine.

Comment (DriesSchaumont) [inline src/concat_text/script.sh:1]: I prefer this:
```suggestion
#!/usr/bin/env bash
```
But not sure if it works on alpine.

File: src/concat_text/config.vsh.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
+name: concat_text
+description: |
```
Comment (DriesSchaumont) [inline src/concat_text/config.vsh.yaml:2]: Do we add author information on this repo?

**Review by DriesSchaumont — CHANGES_REQUESTED** [review]

File: src/concat_text/script.sh (hunk 4 above)
Comment (tverbeiren) [reply src/concat_text/script.sh:1]: Good point, it was there but I changed (and forgot to revert it) while troubleshooting a bug.

File: src/concat_text/config.vsh.yaml (hunk 2 above)
Comment (tverbeiren) [reply src/concat_text/config.vsh.yaml:18]: I suggest to add some points of improvement to an info field or something:

1. What if not all of the input files are zipped or not?
2. Add compression algorithms

WDYT?

Comment (DriesSchaumont) [reply src/concat_text/config.vsh.yaml:18]: Sure, also possible! (Could also just add a YAML comment to the config )

**Review by DriesSchaumont — APPROVED** [review]

**Pull request merged**